### ISSUE TITLE
dist.scm test fails with Guile 3.0.7

### ISSUE THREAD

**Issue by mbakke** [artyom-poptsov/guile-ssh#28]
Hi,

With Guile 3.0.7, `rrepl-get-result, unknown # object error` from `tests/dist.scm` fails like so:

```
Test begin:
  test-name: "rrepl-get-result, unknown # object error"
  source-file: "tests/dist.scm"
  source-line: 185
  source-form: (test-assert "rrepl-get-result, unknown # object error" (begin (set-log-userdata! "rrepl-get-result, unknown # object error") (catch #t (lambda () (call-with-input-string (string-append "scheme@(guile-user)> $4 = " "#<session #<undefined>@#<undefined>:22 (disconnected) 453fff>") rrepl-get-result) #f) (lambda (key . args) (if (equal? key (quote node-repl-error)) (let* ((message (cadr args)) (result (string=? message "Reader error: scm_lreadr: #<unknown port>:1:3: Unknown # object: (#\\<): scheme@(guile-user)> $4 = #<session #<undefined>@#<undefined>:22 (disconnected) 453fff>"))) (unless result (format-log/scm (quote nolog) "rrepl-get-result, unknown # object error" (string-append "Messages do not match: " "expected \"~a\", got \"~a\"") result "Reader error: scm_lreadr: #<unknown port>:1:3: Unknown # object: (#\\<): scheme@(guile-user)> $4 = #<session #<undefined>@#<undefined>:22 (disconnected) 453fff>")) result) (begin (format-log/scm (quote nolog) "rrepl-get-result, unknown # object error" (string-append "Errors do not match: " "expected '~a', got '~a' (args: ~a)") (quote node-repl-error) key args) #f))))))
Test end:
  result-kind: fail
  actual-value: #f
```

**Comment by civodul** [comment]
Hello!

The problem is due to a slightly different error report for `read-error` in Guile 3.0.[67]: the procedure name in the exception was `scm_lreadr` and is now `#f`, and the unknown character is displayed differently (and erroneously, actually).

This patch fixes it for me:

```diff
--- a/tests/dist.scm
+++ b/tests/dist.scm
@@ -183,8 +183,8 @@ Unbound variable: e"
    rrepl-get-result))
 
 (test-error-with-log/= "rrepl-get-result, unknown # object error"
-  'node-repl-error "Reader error: scm_lreadr: #<unknown port>:1:3: \
-Unknown # object: (#\\<): scheme@(guile-user)> \
+  'node-repl-error "Reader error: #f: #<unknown port>:1:3: \
+Unknown # object: (\"#<\"): scheme@(guile-user)> \
 $4 = #<session #<undefined>@#<undefined>:22 (disconnected) 453fff>"
   (call-with-input-string
    (string-append  "scheme@(guile-user)> $4 = "
```

Of course it's a 3.0.7-only fix.  Perhaps the test should use regexps to work with earlier Guile versions as well?

**Comment by artyom-poptsov** [comment]
Hello,

I added a fix in 5cd2ec9 -- now the test uses regular expressions to match the result.

Could you please test it?

Thanks,
avp
